src/http.rs

```diff
@@ -19,23 +27,33 @@ impl HTTP {
         app_name: String,
         instance_id: String,
         authorization: Option<String>,
-    ) -> Result<Self, http_types::Error> {
+    ) -> Result<Self, Error> {
+        #[cfg(feature = "surf")]
+        fn build_header(name: &'static str) -> Result<HeaderName, Error> {
+            HeaderName::from_bytes(name.into())
+        }
+
+        #[cfg(feature = "reqwest")]
+        fn build_header(name: &'static str) -> Result<HeaderName, Error> {
+            Ok(HeaderName::from_static(name))
+        }
+
         Ok(HTTP {
-            client: surf::Client::new(),
+            client: Client::new(),
             app_name,
             instance_id,
             authorization,
-            authorization_header: headers::HeaderName::from_bytes("authorization".into())?,
-            app_name_header: headers::HeaderName::from_bytes("appname".into())?,
-            instance_id_header: headers::HeaderName::from_bytes("instance_id".into())?,
+            authorization_header: build_header("authorization")?,
+            app_name_header: build_header("appname")?,
+            instance_id_header: build_header("instance_id")?,
         })
     }
 
     /// Perform a GET. Returns errors per surf::Client::get.
-    pub fn get(&self, uri: impl AsRef<str>) -> surf::RequestBuilder {
+    pub fn get(&self, uri: impl AsRef<str>) -> RequestBuilder {
         let request = self
             .client
-            .get(uri)
+            .get(uri.as_ref())
```

Comment:
This is needed due to `reqwest` having a different bound in the uri parameter than `surf`.

Reply:
perhaps time to change `impl AsRef<str>` to `&str` and push this back to the caller.